test/goth/runner/probe/test_agent.py

```diff
@@ -0,0 +1,44 @@
+"""Unit tests for probe agents."""
+
+from unittest.mock import MagicMock
+
+from goth.runner.probe import Probe
+from goth.runner.probe.agent import AgentComponent
+
+
+class MockAgentComponent(AgentComponent):
+    """`AgentComponent` implementation used for testing."""
+
+    def start(self, *args, **kwargs):
+        """Abstract method implementation."""
+        pass
+
+    def _init_log_monitor(self):
+        pass
+
+
+def test_agent_duplicate_names():
+    """Test if two agent objects with the same name will be treated as equal."""
```

Comment:
Do we need this property?

Reply:
Please see my comment on `AgentComponent.__hash__`.